components/script/dom/htmlelement.rs

```diff
@@ -560,11 +560,11 @@ static DATA_PREFIX: &str = "data-";
 static DATA_HYPHEN_SEPARATOR: char = '\x2d';
 
 fn is_ascii_uppercase(c: char) -> bool {
-    'A' <= c && c <= 'Z'
+    ('A'..='Z').contains(&c)
 }
 
 fn is_ascii_lowercase(c: char) -> bool {
-    'a' <= c && c <= 'w'
+    c.is_lowercase()
```

Comment:
[`is_lowercase`](https://doc.rust-lang.org/std/primitive.char.html#method.is_lowercase) has different semantics than [`is_ascii_lowercase`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_lowercase). The former will return true even for lowercase non-ASCII characters such as `'δ'` whereas the latter will only check for lowercase characters in the *ASCII*  range (i.e a to z).

Since we want to preserve the semantics, in my previous comment what I was suggesting was to delete the [`fn is_ascii_lowercase` defined in this file](https://github.com/servo/servo/blob/65e0a0c076d2ebd5efb9cf6af887ac167da9594b/components/script/dom/htmlelement.rs#L566) and instead use [`is_ascii_lowercase`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_lowercase) provided by the std library.

For example, in [this line](https://github.com/servo/servo/blob/65e0a0c076d2ebd5efb9cf6af887ac167da9594b/components/script/dom/htmlelement.rs#L604):
```
  if is_ascii_lowercase(next_char) {
      result.push(next_char.to_ascii_uppercase());
  } 
```
we are calling our own definition of `is_ascii_lowercase`. We can instead replace that with a call to the std library version like so:

```
  if next_char.is_ascii_lowercase() {
      result.push(next_char.to_ascii_uppercase());
  } 
```

Similarly, the [definition of `is_ascii_uppercase`](https://github.com/servo/servo/blob/65e0a0c076d2ebd5efb9cf6af887ac167da9594b/components/script/dom/htmlelement.rs#L562) can be deleted and the calls in this file can be replaced with the calls to the std library version.

Reply:
Ohh, I wasn't aware of the difference between them. I will change them.